RockPaperScissors/RockPaperScissors/RockPaperScissors.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  RockPaperScissors.swift
+//  Created by uemu, mireu.
+//  Copyright © yagom academy. All rights reserved.
+//
+
+struct RockPaperScissors {
+    private var nextGame = MukJjiBba()
+    private var turn: CurrentTurn = .user
+    
+    func generateRandomNumber() -> RockPaperScissorsChoice {
+        let randomNumber = Int.random(in: 1...3)
+        
+        switch randomNumber {
+        case 1:
+            return .rock
+        case 2:
+            return .paper
+        case 3:
+            return .scissors
+        default:
+            return .quit
+        }
+    }
+    
+    mutating func compareUserPickAndComputerPick(_ userChoice: RockPaperScissorsChoice, _ computerChoice: RockPaperScissorsChoice) {
+        switch (userChoice, computerChoice) {
+        case (.rock, .scissors), (.paper, .rock), (.scissors, .paper):
+            print("이겼습니다!")
+            turn = .user
+            nextGame.playMukChiBbaGame()
+        case (.rock, .rock), (.paper, .paper), (.scissors, .scissors):
+            print("비겼습니다!")
+            playRockPaperScissors()
+        default:
+            print("졌습니다!")
+            turn = .computer
+            nextGame.playMukChiBbaGame()
+        }
+    }
+    
+    private func selectMenu() -> RockPaperScissorsChoice {
+        let isRunning = true
+        
+        while isRunning {
+            print("가위(1), 바위(2), 보(3)!<종료 : 0> :", terminator: " ")
+            
+            guard let input = readLine(), let userChoice =  RockPaperScissorsChoice(rawValue: Int(input) ?? -1)  else {
+                print("잘못된 입력입니다. 다시 시도해주세요.")
+                continue
+            }
```

Comment:
```suggestion
            guard let input = readLine(),
                  let inputIntegerValue = Int(input),
                  let userChoice =  RockPaperScissorsChoice(rawValue: inputIntegerValue) else {
                print("잘못된 입력입니다. 다시 시도해주세요.")
                continue
            }
```
닐 병합 연산자 대신 이런 방법도 있습니다.

Reply:
닐병합연산자 대신 피드백주신대로 옵셔널바인딩을 사용해 처리를 해봤습니다!

Reply:
닐 병합 연산자를 사용하면 nil인 경우 디폴트값을 주게 되는데 논리적으로 -1을 가지는게 맞지 않다고 판단됩니다.